BraveRewardsUI/SKUs/SKUEnableRewardsViewController.swift

```diff
@@ -0,0 +1,198 @@
+    self.rewards = rewards
+    self.termsURLTapped = termsURLTapped
+    super.init(nibName: nil, bundle: nil)
+    
+    modalPresentationStyle = .overCurrentContext
+    if #available(iOS 13.0, *) {
+      isModalInPresentation = true
+    }
+    transitioningDelegate = self
+  }
+  
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+  
+  var enableRewardsView: SKUEnableRewardsView {
+    return view as! SKUEnableRewardsView // swiftlint:disable:this force_cast
+  }
+  
+  public override func loadView() {
+    view = SKUEnableRewardsView()
+  }
+  
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    enableRewardsView.backgroundView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(tappedBackground)))
+    enableRewardsView.enableRewardsButton.addTarget(self, action: #selector(tappedEnableRewards), for: .touchUpInside)
+    enableRewardsView.termsLabel.onLinkedTapped = { [weak self] _ in
+      self?.termsURLTapped()
+    }
+  }
+  
+  @objc private func tappedBackground() {
+    dismiss(animated: true)
+  }
+  
+  @objc private func tappedEnableRewards() {
+    rewards.ledger.createWalletAndFetchDetails({ _ in })
+    dismiss(animated: true)
+  }
+  
+  override public var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+    if UIDevice.current.userInterfaceIdiom == .phone {
+      return .portrait
+    }
+    return .all
+  }
+  
+  // MARK: - UIViewControllerTransitioningDelegate
+  
+  public func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    return BasicAnimationController(delegate: self, direction: .presenting)
+  }
+  
+  public func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    return BasicAnimationController(delegate: self, direction: .dismissing)
+  }
+}
+
+extension SKUEnableRewardsViewController: BasicAnimationControllerDelegate {
+  func animatePresentation(context: UIViewControllerContextTransitioning) {
+    context.containerView.addSubview(view)
+    view.snp.makeConstraints {
+      $0.edges.equalToSuperview()
+    }
+    view.frame = context.containerView.bounds
+    
+    // Prepare
+    view.layoutIfNeeded()
+    enableRewardsView.backgroundView.alpha = 0.0
+    enableRewardsView.containerView.transform = CGAffineTransform(translationX: 0, y: view.bounds.height - enableRewardsView.containerView.bounds.height)
+    
+    // Animate
+    UIView.animate(withDuration: 0.15) {
+      self.enableRewardsView.backgroundView.alpha = 1.0
+    }
+    UIView.animate(withDuration: 0.4, delay: 0, usingSpringWithDamping: 0.85, initialSpringVelocity: 0, options: [], animations: {
+      self.enableRewardsView.containerView.transform = .identity
+    }, completion: nil)
+    context.completeTransition(true)
```

Comment:
Is this okay? It calls complete transition before the animation is complete.

Reply:
Yeah, I do it early because it enables interaction with the UI as soon as you complete, and we're OK if the view is dismissed immediately upon being able to